core/commands/cat.go

```diff
@@ -20,36 +25,60 @@ it contains.
 	Arguments: []cmds.Argument{
 		cmds.StringArg("ipfs-path", true, true, "The path to the IPFS object(s) to be outputted").EnableStdin(),
 	},
-	Run: func(req cmds.Request) (interface{}, error) {
+	Run: func(req cmds.Request, res cmds.Response) {
 		node, err := req.Context().GetNode()
 		if err != nil {
-			return nil, err
+			res.SetError(err, cmds.ErrNormal)
+			return
 		}
 
 		readers := make([]io.Reader, 0, len(req.Arguments()))
 
-		readers, err = cat(node, req.Arguments())
+		readers, length, err := cat(node, req.Arguments())
 		if err != nil {
-			return nil, err
+			res.SetError(err, cmds.ErrNormal)
+			return
 		}
 
+		res.SetLength(length)
+
 		reader := io.MultiReader(readers...)
-		return reader, nil
+		res.SetOutput(reader)
+	},
+	PostRun: func(res cmds.Response) {
+		if res.Length() < progressBarMinSize {
+			return
+		}
+
+		bar := pb.New(int(res.Length())).SetUnits(pb.U_BYTES)
+		bar.Output = os.Stderr
+		bar.Start()
```

Comment:
Hm, I like that that gives us the ability to redirect the output. But that might be a confusing API since `res.Stdout` in `Run` might be a different `res.Stdout` than in `PostRun`. And since `Response` is an interface, we have to do that as functions :/